cloudvolume/cloudvolume.py

```diff
@@ -161,14 +161,17 @@ def __init__(self, cloudpath, mip=0, bounded=True, autocrop=False, fill_missing=
   @classmethod
   def from_numpy(cls, arr, vol_path='file:///tmp/image/'+generate_random_string(),
                   resolution=(4,4,40), voxel_offset=(0,0,0), 
-                  chunk_size=(128,128,64), layer_type=None, mip_num=1):
+                  chunk_size=(128,128,64), layer_type=None, max_mip=0):
     """
-    mip_num: (int) number of mip levels in the info file
+    max_mip: (int) the maximum mip level id in the info file. 
+    Note that currently the numpy array can only sit in mip 0,
+    the max_mip was only created in info file.
+    the numpy array itself was not downsampled. 
     """
     path = lib.extract_path(vol_path)
 
     if not layer_type:
-      if arr.dtype in (np.uint8, np.float32, np.float16, np.float64):
+      if arr.dtype in (np.bool, np.uint8, np.float32, np.float16, np.float64):
```

Comment:
What about np.intX? They'll creep in eventually, might as well handle it now.

Reply:
I am using np.issubdtype function to include them all!